Robot Code/src/org/usfirst/frc/team3316/robot/commands/turret/TurretPID.java

```diff
@@ -0,0 +1,76 @@
+package org.usfirst.frc.team3316.robot.commands.turret;
+
+import org.usfirst.frc.team3316.robot.Robot;
+import org.usfirst.frc.team3316.robot.commands.DBugCommand;
+
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.PIDOutput;
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+
+public class TurretPID extends DBugCommand
+{
+	private PIDController pid;
+	private double pidOutput;
+
+	public TurretPID() {
+		
+		requires(Robot.turret);
+		
+		pid = new PIDController(0, 0, 0, new PIDSource()
+		{
+			public void setPIDSourceType(PIDSourceType pidSource)
+			{
+			}
+			public double pidGet()
+			{
+				return Robot.turret.getAngle();
+			}
+
+			public PIDSourceType getPIDSourceType()
+			{
+				return PIDSourceType.kDisplacement;
+			}
+		}, new PIDOutput()
+		{
+			public void pidWrite(double output)
+			{
+				pidOutput = output;
+			}
+		});
+		
+		pid.setOutputRange(-1, 1);
+	}
+
+	protected void init()
+	{
+		pid.enable();
+	}
+
+	protected void execute()
+	{
+		pid.setPID((double) config.get("turret_PID_KP"),
+					(double) config.get("turret_PID_KI"),
+					(double) config.get("turret_PID_KD"));
+		
+		pid.setSetpoint((double) config.get("turret_Angle_SetPoint"));
```

Comment:
same question here regarding how we're gonna update the setpoint when running off the camera. see comment on PIDHood.